File: HW2/ship.cpp
//------------------------------------------------------------------------------
// ship.cpp - содержит функции обработки самолета
//------------------------------------------------------------------------------

#include <iostream>
#include "ship.h"
#include <cmath>

//------------------------------------------------------------------------------
// Подсчет рандомного действительного значения
double randomS(double b, double e)
{
    return e + fabs(Random() - b) * 100;
}

//------------------------------------------------------------------------------
// Ввод параметров судна из потока
void Ship::In(std::ifstream &ifst) {
    int type;
    ifst >> type >> this->displacement >> this->speed >> this->range;
    switch (type) {
        case 1:
            this->ship_type = LINER;
            break;
        case 2:
            this->ship_type = TUGBOAT;
            break;
        case 3:
            this->ship_type = TANKER;
            break;
        default:
            std::cerr << "Invalid transport type number" << std::endl;
    }
}

// Случайный ввод параметров судна
void Ship::InRnd() {
    this->range = randomS(1.54, 100);
    this->speed = Random();
    this->displacement = Random();
    switch (Random() % 3) {
        case 0:
            this->ship_type = LINER;
            break;
        case 1:
            this->ship_type = TUGBOAT;
            break;
        case 2:
            this->ship_type = TANKER;
            break;
    }
}

//------------------------------------------------------------------------------
// Вывод параметров судна в поток
void Ship::Out(std::ofstream &ofst) {
    ofst << "Ship: ship type = ";
    switch (this->ship_type) {
        case LINER:
            ofst << "liner";
            break;
        case TUGBOAT:
            ofst << "tugboat";
            break;
        case TANKER:
            ofst << "tanker";
            break;
    }
    ofst << " Ship specifications: displacement = "
         << this->displacement
         << ", Speed = " << this->speed
         << ", Range = " << this->range
         << ". Ideal time of passage = " << this->CountTime() << "\n";
}


//------------------------------------------------------------------------------
// Идеальное время прохождения пути для судна

double Ship::CountTime() {
    double time = this->range / (double)this->speed;
    return time;
}

File: HW3/src/ship.py
import newRandom
from enum import Enum


# ----------------------------------------------

class ShipType(Enum):
    UNKNOWN_TYPE = 0
    LINER = 1
    TUGBOAT = 2
    TANKER = 3


# Класс корабля
class Ship:
    # Конструктор корабля
    def __init__(self):
        self.range = 0
        self.speed = 0
        self.displacement = 0
        self.ship_type = ShipType.UNKNOWN_TYPE.name

    # Считывание параметров поезда из строки
    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        type = int(strArray[i])
        if 1 <= type <= 3:
            self.ship_type = ShipType(type).name
        else:
            self.ship_type = ShipType.UNKNOWN_TYPE.name
        self.displacement = int(strArray[i + 1])
        self.speed = int(strArray[i + 2])
        self.range = int(strArray[i + 3])
        i += 4
        return i

    # Случайная генерация параметров корабля
    def RandomIn(self):
        self.range = newRandom.randDouble(10, 10000)
        self.speed = newRandom.randInt(100, 10000)
        self.displacement = newRandom.randInt(1, 1000)
        self.ship_type = ShipType(newRandom.randInt(1, 3)).name

    # Вывод данных корабля в файл
    def Write(self, ostream):
        ostream.write("Ship: type = {}, displacement = {} ,speed = {}, range = {}, Ideal time of travel = {}"
                      .format(self.ship_type, self.displacement, self.speed, self.range, self.CountTime()))
        pass

    #  Идеальное время прохождения пути для корабля
    def CountTime(self):
        return self.range / self.speed
